### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by aalexcomas11**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,17 +1,45 @@
...
+  console.log(err)
+})
 
-app.use(cors())
 app.use(bodyParser())
-app.use(sampleAPI.routes())
+app.use(cors())
+
+//authorization
+require('./auth')
```
Comment (Jehfkemsy) [inline src/app.ts]: is there a reason why this import can't be at the top of the file?

File: src/test/api.test.ts [hunk 2]
```diff
@@ -1,11 +1,106 @@
...
+  beforeAll(() => {
+    api = request(app.listen())
+    messageBus = createMessageBus()
+    userMap = new Map<string, any>()
+    userMap.set(user1.email, user1)
+    userMap.set(user2.email, user2)
+    userMap.set(user3.email, user3)
+  })
+
+  it('user without invalid email cannot create account', async () => {
```
Comment (Jehfkemsy) [inline src/test/api.test.ts]: think you meant to write "with" instead of "without"

File: src/Services/AuthenticationService.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+interface IAuthenticationService {
+  signUp: (ctx: any) => Promise<IUser>
+  login: (ctx: any) => Promise<string>
+}
+
+const authenticationStore = new AuthenticationStore()
+const messageBus = createMessageBus()
+
+export default class AuthenticationService implements IAuthenticationService {
+  async signUp(ctx: any): Promise<IUser> {
```
Comment (Jehfkemsy) [inline src/Services/AuthenticationService.ts]: consider adding steps that disable user account until their email address has been verified

Comment (aalexcomas11) [reply src/Services/AuthenticationService.ts]: I’ll add it to the next sprint.

File: src/app.ts (hunk 1 above)
Comment (aalexcomas11) [reply src/app.ts]: The order of the middleware matters, the Auth middleware is dependent on everything on top of it.

File: src/test/api.test.ts (hunk 2 above)
Comment (aalexcomas11) [reply src/test/api.test.ts]: Yes, I’ll change it

File: src/app.ts (hunk 1 above)
Comment (Jehfkemsy) [reply src/app.ts]: 👍

**Pull request merged**